drift-integration-tests/src/test/java/io/airlift/drift/integration/guice/ThrowingServiceHandler.java

```diff
@@ -15,11 +15,20 @@
  */
 package io.airlift.drift.integration.guice;
 
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import static com.google.common.util.concurrent.Futures.immediateFuture;
 import static java.lang.Math.toIntExact;
 
 public class ThrowingServiceHandler
         implements ThrowingService
 {
+    @GuardedBy("this")
```

Comment:
Given the usage pattern, we could make this final and initialize the future here. Alternatively, should we clear the future field in `release()` (so the service could be called multiple times)?

Reply:
> Alternatively, should we clear the future field 

+1